astroid/brain/brain_namedtuple_enum.py

```diff
@@ -544,18 +544,49 @@ def infer_typing_namedtuple(node, context=None):
     return infer_named_tuple(node, context)
 
 
+def _is_enum_subclass(cls: astroid.ClassDef) -> bool:
+    """Return whether cls is a subclass of an Enum.
+
+    Warning: For efficiency purposes, this function immediately returns False if enum hasn't
+    been imported in the module of the ClassDef. This means it fails to detect an Enum
+    subclass that is imported from a new module and subclassed, e.g.
```

Comment:
I think we need this kind of compromise if we want to keep pylint reasonably fast. But on the other hand I think that what pylint bring is a lot of checks in a lot more time than what flake8 does for example. I guess if we try to make pylint's fast and compromise on correctness pylint will stay slow and only be a little less wrong. So I'm a little torn here. How slow would the check be is we check the ancestors of the class (Knowing that we're going to cache the ancestors in #1120) ?

Reply:
Yeah, looking at this a bit more I think this is a case of premature optimization on my part. I ran pylint on this repository with `astroid==2.6.6`, then with this version, and then with this version but the early return commented out, and didn't see a noticeable difference, they were all in the ballpark for 19-20 seconds. In general it seems there are more important performance issues with pylint that your team is working on.

Especially with the ancestors being cached (very cool!), and the fact that `mro` is almost certainly going to be called at *some* point when using pylint since it's essential for E1101 (no-member), I don't think the extra check is worth having. And certainly not at the expense of weird false positives!

So if you agree, I'll update this PR to remove the check from the code.

Reply:
Thank you for taking the time to do the comparison. Let's use the ancestors then :) We do have major performance issue to tackle. I'm even thinking of a new checker freeze to be able to focus on performance, because it's hard to find time for that.